File: Tema 10/ListadoPersonasClean/Domain/UseCases/DepartamentoUseCase.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Domain.Entities;
using Domain.Interfaces;
using Domain.Repositorios;

namespace Domain.UseCases
{
    public class DepartamentoUseCase : IDepartamentoUseCase
    {
        private readonly IDepartamentoRepositorio _departamentoRepositorio;

        public DepartamentoUseCase(IDepartamentoRepositorio departamentoRepositorio)
        {
            _departamentoRepositorio = departamentoRepositorio;
        }

        public List<Departamento> getDepartamentos()
        {
            return _departamentoRepositorio.getDepartamentos();
        }

        public Departamento getDepartamento(int id)
        {
            return _departamentoRepositorio.getDepartamento(id);
        }

        public int agregarDepartamento(Departamento departamento)
        {
            return _departamentoRepositorio.agregarDepartamento(departamento);
        }

        public int actualizarDepartamento(Departamento departamento)
        {
            return _departamentoRepositorio.actualizarDepartamento(departamento);
        }

        public int eliminarDepartamento(int id)
        {
            return _departamentoRepositorio.eliminarDepartamento(id);
        }
    }
}

File: Tema15/ListadoPersonasAngular/src/Domain/UseCases/DepartamentoUseCase.ts
import { inject, injectable } from 'inversify';
import { IDepartamentoUseCase } from '../Interfaces/IDepartamentoUseCase';
import { IDepartamentoRepository } from '../Repositories/IDepartamentoRepository';
import { Departamento } from '../Entities/Departamento';
import { TYPES } from '../../Core/Types';

@injectable()
export class DepartamentoUseCase implements IDepartamentoUseCase {
  constructor(
    @inject(TYPES.IDepartamentoRepository) private departamentoRepository: IDepartamentoRepository
  ) {}

  async getDepartamentos(): Promise<Departamento[]> {
    return await this.departamentoRepository.getDepartamentos();
  }

  async getDepartamento(id: number): Promise<Departamento> {
    return await this.departamentoRepository.getDepartamento(id);
  }

  async agregarDepartamento(departamento: Departamento): Promise<number> {
    return await this.departamentoRepository.agregarDepartamento(departamento);
  }

  async actualizarDepartamento(departamento: Departamento): Promise<number> {
    return await this.departamentoRepository.actualizarDepartamento(departamento);
  }

  async eliminarDepartamento(id: number): Promise<number> {
    return await this.departamentoRepository.eliminarDepartamento(id);
  }
}